conftest.py

```diff
@@ -4,11 +4,38 @@
 import requests
 from typing import Callable
 import logging
+from dataclasses import dataclass
 
 
 log = logging.getLogger(__name__)
 
 
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int  # RuntimeError: Sensor didn't reset to factory property if float
+
+
+    def __post_init__(self):
+        if not isinstance(self.name, str) or self.name == "":
+            raise TypeError("'name' is not a string or empty")
+
+        if not isinstance(self.hid, str) or self.hid == "":
+            raise TypeError("'hid' is not a string or empty")
+
+        if not isinstance(self.model, str) or self.model == "":
+            raise TypeError("'model' is not a string or empty")
+
+        if not isinstance(self.firmware_version, int) or self.firmware_version == "":
+            raise TypeError("'firmware_version' is not an int or empty")
+
+        if not isinstance(self.reading_interval, int) or self.reading_interval == "":
+            raise TypeError("'reading_interval' is not a float or empty")
```

Comment:
This error message should say `int` instead of `float`.

Reply:
fixed